astropy/io/fits/tests/test_image.py

```diff
@@ -1125,6 +1128,21 @@ def test_hdu_creation_with_scalar(self):
         with pytest.raises(TypeError, match=msg):
             fits.PrimaryHDU(data=np.array(1))
 
+    def test_section_cutout2d_partial(self):
+        coord = SkyCoord("0.04694018 75.2179368", unit="deg")
+
+        with fits.open(self.data("compressed_2d_float.fits")) as hdul:
+            section = hdul[1].section
+            wcs = WCS(hdul[1].header)
+
+            try:
+                cutout = Cutout2D(
+                    section, coord, 800, wcs=wcs, mode="partial", copy=True
+                )
+                assert cutout is not None
+            except IndexError as e:
+                pytest.fail(f"Test failed with error: {e}")
```

Comment:
Why do we need special `pytest.fail` here instead of letting the original traceback bubble up for debugging?